pkg/kubelet/network/hairpin/hairpin.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hairpin
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	osexec "os/exec"
+	"path"
+	"regexp"
+	"strconv"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/util/exec"
+)
+
+const (
+	sysfsNetPath            = "/sys/devices/virtual/net"
+	hairpinModeRelativePath = "brport/hairpin_mode"
+	hairpinEnable           = "1"
+)
+
+var (
+	ethtoolOutputRegex = regexp.MustCompile("peer_ifindex: (\\d+)")
+)
+
+type plugin struct {
+	e exec.Interface
+}
+
+func New() *plugin {
```

Comment:
rather than this, why not do:

```
func SetUpContainer(containerPid int, containerInterfaceName string) error {
    e := exec.New()
    return setUpContainerInternal(e, containerPid, containerInterfaceName)
}

func setUpContainerInternal(e exec.Interface, containerPid int, containerInterfaceName string) error {
   hostIfName, err := p.findPairInterfaceOfContainerInterface(e, containerPid, containerInterfaceName)
   //...
}
```

Then write tests to `setUpContainerInternal()` - the caller should not see this New() business..